pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -202,6 +204,18 @@ func destroyVdiskVolume(ctx *volumemgrContext, status types.VolumeStatus) (bool,
 		return created, "", errors.New(errStr)
 	}
 	if info.Mode()&os.ModeDevice != 0 {
+		if err := tgt.VHostDeleteIBlock(status.Key(), status.WWN); err != nil {
+			errStr := fmt.Sprintf("Error deleting vhost for %s, error=%v",
+				status.Key(), err)
+			log.Error(errStr)
+			return created, "", errors.New(errStr)
```

Comment:
If you ever hit this error case you will never delete the Target, right. 

Might make sense to call TargetDeleteIBlock() here and ignore any error.

Reply:
Good point, I removed returns